components/calendar/src/provider/chinese_based.rs

```diff
@@ -88,18 +88,34 @@ impl PackedChineseBasedYearInfo {
     /// could occur after the Winter Solstice if the solstice is pinned to December 20.
     const FIRST_NY: i64 = 18;
 
+    /// out_of_valid_astronomical_range is true when the data is for a date that is well
+    /// outside calendrical_calculations::chinese_based::WELL_BEHAVED_ASTRONOMICAL_RANGE.
+    /// It clamps some values to avoid debug assertions on calendrical invariants.
     pub(crate) fn new(
         month_lengths: [bool; 13],
         leap_month_idx: Option<u8>,
         ny_offset: i64,
+        out_of_valid_astronomical_range: bool,
     ) -> Self {
+        // This assertion is an API correctness assertion and even bad calendar arithmetic
+        // should not produce this
         debug_assert!(
             !month_lengths[12] || leap_month_idx.is_some(),
             "Last month length should not be set for non-leap years"
         );
-        let ny_offset = ny_offset - Self::FIRST_NY;
-        debug_assert!(ny_offset >= 0, "Year offset too small to store");
-        debug_assert!(ny_offset < 34, "Year offset too big to store");
+        let mut ny_offset = ny_offset - Self::FIRST_NY;
+
+        if out_of_valid_astronomical_range {
```

Comment:
should this be `#[cfg(debug_assertions)]`, along with the `out_of_valid_astronomical_range` parameter?

Reply:
In that case you can always clamp, which is only one AND instruction, instead of computing the flag and branching.